src/main/kotlin/uk/gov/dluhc/printapi/service/S3PhotoService.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.dluhc.printapi.service
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.core.exception.SdkException
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class S3PhotoService(
+    private val s3Client: S3Client
+) {
+    /**
+     * Removes the photo that is on the printed "Elector Document" (i.e. Certificate/AED) from S3.
+     */
+    fun removePhoto(photoS3Arn: String) {
+        try {
+            with(parseS3Arn(photoS3Arn)) {
+                s3Client.deleteObject(DeleteObjectRequest.builder().bucket(bucket).key(path).build())
+                logger.info { "Deleted photo with S3 arn [$photoS3Arn]" }
+            }
+        } catch (e: SdkException) {
+            logger.warn { "Unable to delete photo with S3 arn [$photoS3Arn] due to error [${e.cause?.message ?: e.cause}]" }
```

Comment:
I'm going to change this as there are probably cases where we want to throw an exception and rollback the transaction in the `CertificateDataRetentionService` (for example when there's a server error).

I thought it makes sense to be forgiving in case the photo cannot be found (although there's no mention of an exception being thrown in this scenario in the `S3Client` javadocs). I'll do some more investigation.

If the photo cannot be found, then it makes sense to continue to remove the data from the database, but otherwise we should probably roll back the transaction. However, this would mean the transaction is rolled back for all certificates in the list, which may not be right. Hence why I'd like to give more thought to the transaction boundaries. Suggestions welcomed 😄